demisto_sdk/commands/validate/validate_manager.py

```diff
@@ -0,0 +1,841 @@
+from demisto_sdk.commands.common.hook_validations.mapper import MapperValidator
+from demisto_sdk.commands.common.hook_validations.pack_unique_files import \
+    PackUniqueFilesValidator
+from demisto_sdk.commands.common.hook_validations.playbook import \
+    PlaybookValidator
+from demisto_sdk.commands.common.hook_validations.readme import ReadMeValidator
+from demisto_sdk.commands.common.hook_validations.release_notes import \
+    ReleaseNotesValidator
+from demisto_sdk.commands.common.hook_validations.reputation import \
+    ReputationValidator
+from demisto_sdk.commands.common.hook_validations.script import ScriptValidator
+from demisto_sdk.commands.common.hook_validations.structure import \
+    StructureValidator
+from demisto_sdk.commands.common.hook_validations.widget import WidgetValidator
+from demisto_sdk.commands.common.tools import (checked_type,
+                                               filter_packagify_changes,
+                                               find_type, get_pack_name,
+                                               get_remote_file, get_yaml,
+                                               is_file_path_in_pack,
+                                               run_command)
+
+
+class FileType(Enum):
+    integration = 1
+    script = 2
+    playbook = 3
+    testplaybook = 4
+    betaintegration = 5
+    incidentfield = 6
+    indicatorfield = 7
+    reputation = 8
+    layout = 9
+    dashboard = 10
+    incidenttype = 11
+    mapper = 12
+    classifier_5_9_9 = 13
+    classifier = 14
+    widget = 15
+    report = 16
+    connection = 17
+    readme = 18
+    releasenotes = 19
+
+
+SKIPPED_FILE_TYPES = ('changelog', 'description', 'testplaybook')
+
+
+class ValidateManager:
+    def __init__(self, is_backward_check=True, prev_ver=None, use_git=False, only_committed_files=False,
+                 print_ignored_files=False, skip_conf_json=True, validate_id_set=False, file_path=None,
+                 validate_all=False, is_external_repo=False, skip_pack_rn_validation=False, print_ignored_errors=False,
+                 silence_init_prints=True):
+        self.handle_error = BaseValidator(print_as_warnings=print_ignored_errors).handle_error
+        self.print_ignored_errors = print_ignored_errors
+        self.skip_docker_checks = False
+        self.skip_conf_json = skip_conf_json
+        self.is_backward_check = is_backward_check
+        self.validate_id_set = validate_id_set
+        self.is_circle = only_committed_files
+        self.validate_all = validate_all
+        self.file_path = file_path
+        self.branch_name = ''
+        self.use_git = use_git
+        self.skip_pack_rn_validation = skip_pack_rn_validation
+        self.changes_in_schema = False
+        self.check_only_schema = False
+        self.prev_ver = prev_ver if prev_ver else 'origin/master'
+        self.print_ignored_files = print_ignored_files
+        self.added_rn = set()
+        self.ignored_files = set()
```

Comment:
seperate this to sections, 

General configuration....
Error configurations...
Git handling..